plugin/pkg/scheduler/equivalence_cache.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduler
+
+import (
+	"github.com/golang/groupcache/lru"
+	"hash/adler32"
+
+	"k8s.io/kubernetes/pkg/api"
+	hashutil "k8s.io/kubernetes/pkg/util/hash"
+	"k8s.io/kubernetes/pkg/util/sets"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/algorithm"
+	schedulerapi "k8s.io/kubernetes/plugin/pkg/scheduler/api"
+	"sync"
+)
+
+const maxCacheEntries = 4096
+
+type HostPredicate struct {
+	Fit         bool
+	FailReasons []algorithm.PredicateFailureReason
+}
+
+type AlgorithmCache struct {
+	predicatesCache *lru.Cache
+	prioritiesCache *lru.Cache
```

Comment:
Priorities are much different from predicates (as I wrote in the other PR). In particular, a bunch of them are actually "global" so any change in the cluster can potentially affect scores for all nodes.
That said we should first implement: https://github.com/kubernetes/kubernetes/issues/24246
[though the initial motivation of that was different] and then cache results of "map" phase of priority functions and always recompute reduce-phase from those values.

I already started working on that issue - the initial PR is here:
https://github.com/kubernetes/kubernetes/pull/31606
Once this is merged, the remaining thing is just to switch all the priority functions to the new framework, which shouldn't be that hard.